wurst/objects/units/ShopObjects.wurst

```diff
@@ -2,35 +2,64 @@ package ShopObjects
 
 import Assets
 import UnitObjEditing
+import AbilityObjEditing
+import ObjEditingUtils
+import ObjectIdGenerator
+
+import LinkedList
 
 import LocalObjectIDs
+import LocalAssets
 
 public function HeroDefinition.setTotemShopInterface()
     this
-    ..setLumberCost(0)
-    ..setFoodCost(0)
-    ..setGoldCost(0)
-    ..setStockStartDelay(0)
-    ..setStockReplenishInterval(999) // Will be programatically refilled
-    ..setStockMaximum(1)
+        ..setLumberCost(0)
+        ..setFoodCost(0)
+        ..setGoldCost(0)
+        ..setStockStartDelay(0)
+        ..setStockReplenishInterval(999) // Will be programatically refilled
+        ..setStockMaximum(1)
 
 @compiletime function createDummies()
     new HeroDefinition(UNIT_RANDOM_TROLL, UnitIds.shadowhunter)
-    ..setTotemShopInterface()
-    ..setName("Random Troll |cffffcc00(D)|r")
-    ..setTooltipBasic("Random")
-    ..setTooltipExtended("")
-    ..setHotkey("D")
-    ..setStockReplenishInterval(0)
-    ..setIconGameInterface(Icons.bTNSelectHeroOn)
-    ..setUnitSoundSet("")
+        ..setTotemShopInterface()
+        ..setName("Random Troll |cffffcc00(D)|r")
+        ..setTooltipBasic("Random")
+        ..setTooltipExtended("")
+        ..setHotkey("D")
+        ..setStockReplenishInterval(0)
+        ..setIconGameInterface(Icons.bTNSelectHeroOn)
+        ..setUnitSoundSet("")
 
     new HeroDefinition(UNIT_REPICK_TROLL, UnitIds.shadowhunter)
-    ..setTotemShopInterface()
-    ..setName("Repick Troll |cffffcc00(R)|r")
-    ..setTooltipBasic("Repick")
-    ..setTooltipExtended("")
-    ..setHotkey("R")
-    ..setStockReplenishInterval(0)
-    ..setIconGameInterface(Icons.bTNCancel)
-    ..setUnitSoundSet("")
+        ..setTotemShopInterface()
+        ..setName("Repick Troll |cffffcc00(R)|r")
+        ..setTooltipBasic("Repick")
+        ..setTooltipExtended("")
+        ..setHotkey("R")
+        ..setStockReplenishInterval(0)
+        ..setIconGameInterface(Icons.bTNCancel)
+        ..setUnitSoundSet("")
+
+@compiletime function createTotem()
+    let ABILITY_SELECT_UNITS_TOTEM = compiletime(UNIT_ID_GEN.next())
+
+    new AbilityDefinitionAlliedBuilding(ABILITY_SELECT_UNITS_TOTEM)
+        ..setActivationRadius(1, 1) // Small activation radius so it doesn't try selecting trolls
+        ..setShowSelectUnitButton(1, false)
+
+    new BuildingDefinition(UNIT_TROLL_TOTEM, UnitIds.cantina)
+        ..setName("Troll Totem")
+        ..setDescription("Troll picker")
```

Comment:
Does this have any functional purpose?

Reply:
I actualyl dont know

Reply:
Didn't you add it? It wasn't in the raw definition.

Reply:
I copied this from another BuildingDefinition iirc, I can remove if u want

Reply:
This field correspond to the text displayed when you hoover your icon over the building icon on your command button panel.

Reply:
I thought that required selecting multiple units though. You can't do that for the totem.

Reply:
can't you just select the totem and your troll after you picked?